### DISCUSSION TITLE
How to define MonitoredItemService in asyncua Server

### DISCUSSION THREAD

**Discussion by dibyendumca** [FreeOpcUa/opcua-asyncio#1488] [Q&A]
Hello,

I need to define some node which should trigger data change notification only when its value is a specific value.
How to implement that in the Server/Client side with DatachangeFilter and AggregateFilter?
Can anyone help me out?

Thanks

**Comment by schroeder-** [comment]
Don't think there is any way to do this. Only a Deadband is possible, which is not what you want.

Why not filter out the values in our subscription handler?

**Reply by dibyendumca** [reply]
Hello Alexander,
Thanks for your comment...
What is this MonitoredItemService used for?
And can you please provide some sample code of doing it as you said in the Subscription handler?
Thanks

**Reply by schroeder-** [reply]
```
class SubscriptionHandler:
    limit = {}
    """
    The SubscriptionHandler is used to handle the data that is received for the subscription.
    """
    def datachange_notification(self, node: Node, val, data):
        """
        Callback for asyncua Subscription.
        This method will be called when the Client received a data change message from the Server.
        """
        if limit[node] == val:
           print(f'value limit reached {node} {val}')

...

        handler = SubscriptionHandler()
        # We create a Client Subscription.
        subscription = await client.create_subscription(500, handler)
        var = await client.nodes.objects.get_child([f"{idx}:MyObject", f"{idx}:MyVariable"])
        handler.limit[var] = 5.0  # error value
        await subscription.subscribe_data_change(var)
```

**Reply by dibyendumca** [reply]
Thank you so much......
I shall try it out, but still I am curious to know about this MonitoredItemService class.
How is the MonitoredItemService used in Server / Client?
I mean to ask, What this MonitoredItemService is used for?
Thanks

**Reply by schroeder-** [reply]
What you mean? The MonitoredItemService class is used by the server to handle subscriptions for clients.

**Reply by dibyendumca** [reply]
Hello,
OK, I see, Thanks...
I see a feature as described in OPC UA Foundation Website.
![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/61493364/1815684f-df25-4fea-878f-11784f22eb99)

How shall I implement this DISCARDING OF OLDEST VALUE feature in asyncua?
Please provide some example.

Thanks

**Reply by schroeder-** [reply]
I think this is not handled in our server.

**Reply by dibyendumca** [reply]
Will it be available in the next releases?

**Reply by schroeder-** [reply]
I don't think so,  feel free to implement it and create a pr.
